src/webots/nodes/WbRobot.cpp

```diff
@@ -190,11 +191,35 @@ void WbRobot::preFinalize() {
 
   if ((WbTokenizer::worldFileVersion() < WbVersion(2020, 1, 0) ||
        (proto() && proto()->fileVersion() < WbVersion(2020, 1, 0))) &&
-      mControllerArgs->value().size() == 1 && mControllerArgs->value()[0].contains(" ")) {
-    mControllerArgs->setValue(mControllerArgs->value()[0].split(" "));
-    parsingWarn(tr("Robot.controllerArgs data type changed from SFString to MFString in Webots R2020b. "
-                   "Splitting controllerArgs at space boundaries. "
-                   "You may need to update your PROTO and/or world file(s)."));
+      mControllerArgs->value().size() == 1 && mControllerArgs->value()[0].contains(' ')) {
+    // we need to split the controllerArgs[0] at space boundaries into a list of strings
+    // taking into account quotes and double quotes to avoid splitting in the middle of a quoted (or double quoted) string
+    QStringList arguments;
+    const QString args = mControllerArgs->value()[0].trimmed();
+    const int l = args.length();
+    bool doubleQuote = false;
+    bool singleQuote = false;
```

Comment:
I would add the `inside` prefix to make these two variables more self-explanatory.